src/tests/ov_helpers/ov_snippets_models/src/subgraph_fq.cpp

```diff
@@ -59,6 +59,19 @@ std::shared_ptr<ov::Model> ThreeFQFunction::initReference() const {
                                                                               std::vector<float>{127},
                                                                               ov::element::i8);
     auto fq1 = ov::builder::subgraph::makeFakeQuantizeTypeRelaxed(fq0, ov::element::f32, fq1_data);
+// ARM has enough gprs to tokenize three fq into one subgraph
+#if defined(OPENVINO_ARCH_ARM64)
+    auto fq2_data = ov::builder::subgraph::FakeQuantizeOnDataWithConstant(256,
+                                                                              std::vector<ov::Shape>{{1, 3, 1, 1}, {1, 3, 1, 1}, {1}, {1}},
+                                                                              std::vector<float>{-98.16883, -142.82466, -155.0642700},
+                                                                              std::vector<float>{97.334106, 141.599884, 153.8145446},
+                                                                              std::vector<float>{0},
+                                                                              std::vector<float>{255},
+                                                                              ov::element::u8);
+    auto fq2 = ov::builder::subgraph::makeFakeQuantizeTypeRelaxed(fq1, ov::element::f32, fq2_data);
+    auto subgraph1 = std::make_shared<ov::snippets::op::Subgraph>(NodeVector{data0},
+                                      std::make_shared<ov::Model>(NodeVector{fq2}, ParameterVector{indata0}));
+#else
```

Comment:
Could you please clarify why you added this change if arm doesn't support FQ in snippets for now?

Reply:
You are right that arm doesn't support FQ in snippets for now. Yet smoke_Snippets_ThreeFQFunction https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/tests/src/pass/collapse_subgraph.cpp#L103-L108 verifies TokenizeSnippets https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/tests/src/pass/collapse_subgraph.cpp#L21. And TokenizeSnippets is plugin independent, where the function `is_supported_fq_op` says fq is supported. And since now arm has more grps, the behavior here is different, https://github.com/openvinotoolkit/openvino/blob/master/src/common/snippets/src/pass/collapse_subgraph.cpp#L614. The three fq nodes can be fully tokenized into one subgraph on arm. So the ref subgraph in test should align with it. Further disccussion are always welcomed. Thanks Alexandra!

Reply:
Ok, I see. Thanks for the explanation!

Now we set the GPR count on backend side. Since this test is common test - Snippets don't have any common limitations for data count now. Thus, I'd expect here your new solution instead of previous. I mean if Snippets support unlimited data count in common pipeline now, this test should tokenize all FQs to one Subgraph as you added. Let's leave only your suggestion here.

P.S. My major comment of the PR - is using of `#if defined(OPENVINO_ARCH_ARM64)` in common parts. Snippets and snippets tests should have `#if defined(OPENVINO_ARCH_ARM64)` in the code.

Reply:
Applied. Thanks Alexandra!